File: app/native/platform/src/backend/backend_stub.zig
// ArchoeraMusic 平台能力原生桥接
// Copyright (C) 2026 Archoera && BetaStudio2
// SPDX-License-Identifier: AGPL-3.0-or-later

//! P0 通用 stub 后端：能力位图恒 0，一切调用返回 UNSUPPORTED。
//!
//! P1–P5 逐平台落地后（见 docs/platform-native-bridge.md §7），各平台后端
//! 覆盖对应函数；本文件仅兜底未知目标平台与编译期无后端的场景。

const core = @import("../core.zig");

pub const Backend = struct {
    pub fn caps() u32 {
        return 0;
    }
    pub fn init() i32 {
        return core.OK;
    }
    pub fn shutdown() i32 {
        return core.OK;
    }
    pub fn powerSetSleepInhibit(on: i32) i32 {
        _ = on;
        return core.ERR_UNSUPPORTED;
    }
    pub fn powerSetScreenEvents(on: i32) i32 {
        _ = on;
        return core.ERR_UNSUPPORTED;
    }
    pub fn windowSetEvents(on: i32) i32 {
        _ = on;
        return core.ERR_UNSUPPORTED;
    }
    pub fn mediaSetTrack(meta: ?*const core.TrackMeta) i32 {
        _ = meta;
        return core.ERR_UNSUPPORTED;
    }
    pub fn mediaSetPlayback(state: i32, position_ms: i64, speed: f64, volume: f64, loop: i32, shuffle: i32) i32 {
        _ = .{ state, position_ms, speed, volume, loop, shuffle };
        return core.ERR_UNSUPPORTED;
    }
    pub fn mediaSetWindow(win: i64) i32 {
        _ = win;
        return core.OK;
    }
    pub fn appInstanceAcquire() i32 {
        return 1;
    }
    pub fn notify(title: []const u8, body: []const u8) i32 {
        _ = .{ title, body };
        return core.OK;
    }
    pub fn systemAccent() ?[3]u8 {
        return null;
    }
    pub fn systemAccentSetEvents(on: i32) i32 {
        _ = on;
        return core.ERR_UNSUPPORTED;
    }
};

pub fn appInstanceAcquire() i32 {
    return 1; // 无平台：不阻断
}


pub fn notify(title: []const u8, body: []const u8) i32 {
    _ = title; _ = body;
    return core.OK;
}

pub fn systemAccent() ?[3]u8 {
    return null;
}

File: app/native/platform/src/backend_stub.cpp
// ArchoeraMusic 平台能力原生桥接
// Copyright (C) 2026 Archoera && BetaStudio2
// SPDX-License-Identifier: AGPL-3.0-or-later

//! 未实现平台的兜底后端：能力位图 0，所有调用返回 UNSUPPORTED。
//! 保证 ABI 在任意目标可链接、可运行（Dart 侧据此降级）。

#include "backend.h"

#include "core.h"

namespace archoera {

uint32_t caps() { return 0; }

int32_t init() { return OK; }
int32_t shutdown() { return OK; }

int32_t powerSetSleepInhibit(int32_t) { return ERR_UNSUPPORTED; }
int32_t powerSetScreenEvents(int32_t) { return ERR_UNSUPPORTED; }

int32_t windowSetEvents(int32_t) { return ERR_UNSUPPORTED; }

int32_t mediaSetTrack(const AplTrackMeta*) { return ERR_UNSUPPORTED; }
int32_t mediaSetPlayback(int32_t, int64_t, double, double, int32_t, int32_t) {
    return ERR_UNSUPPORTED;
}
int32_t mediaSetWindow(int64_t) { return ERR_UNSUPPORTED; }

int32_t appInstanceAcquire() { return 1; }

bool systemAccent(int32_t*, int32_t*, int32_t*) { return false; }
int32_t systemAccentSetEvents(bool) { return ERR_UNSUPPORTED; }
int32_t systemThemeSetEvents(bool) { return ERR_UNSUPPORTED; }

int32_t notify(const char*, const char*) { return ERR_UNSUPPORTED; }

// ArchoeraOS 会话：仅 Linux/Wayland 提供，其余平台恒不支持。
int32_t osSessionSetEvents(int32_t) { return ERR_UNSUPPORTED; }
int32_t osSessionSetBrightness(int32_t) { return ERR_UNSUPPORTED; }
int32_t osSessionSetVolume(int32_t) { return ERR_UNSUPPORTED; }
int32_t osSessionSetScreenEnabled(int32_t) { return ERR_UNSUPPORTED; }
int32_t osSessionPowerOff() { return ERR_UNSUPPORTED; }
int32_t osSessionReboot() { return ERR_UNSUPPORTED; }
int32_t osSessionSuspend() { return ERR_UNSUPPORTED; }
int32_t osSessionHibernate() { return ERR_UNSUPPORTED; }
int32_t osSessionSetOutputScale(int32_t) { return ERR_UNSUPPORTED; }
int32_t osSessionSetOutputMode(int32_t, int32_t) { return ERR_UNSUPPORTED; }
int32_t osSessionSetOutputTransform(int32_t) { return ERR_UNSUPPORTED; }
int32_t osSessionKey(int32_t, int32_t) { return ERR_UNSUPPORTED; }

}  // namespace archoera

// ── 系统资源 / 蓝牙：未覆盖平台返回 UNSUPPORTED ─────────────────────
namespace sysinfo {
bool statsAvailable() { return false; }
bool bluetoothAvailable() { return false; }
void probe() {}
int32_t stats(AplSysStats*) { return ERR_UNSUPPORTED; }
int32_t btState(AplBtState*) { return ERR_UNSUPPORTED; }
void shutdown() {}
}  // namespace sysinfo
